platform_plugin_ontask/datasummary/backends/completion.py

```diff
@@ -0,0 +1,42 @@
+"""Data summary for completion data."""
+
+from collections import defaultdict
+
+from opaque_keys.edx.keys import CourseKey
+
+from platform_plugin_ontask.datasummary.backends.base import DataSummary
+from platform_plugin_ontask.edxapp_wrapper.completion import CompletionService
+from platform_plugin_ontask.edxapp_wrapper.enrollments import get_user_enrollments
+from platform_plugin_ontask.utils import get_course_units
+
+
+class CompletionDataSummary(DataSummary):
+    """Data summary for completion data."""
+
+    def get_data_summary(self) -> dict:
+        """
+        Get the completion data summary.
+
+        Returns:
+            dict: A dataframe with the completion data summary
+        """
+        course_key = CourseKey.from_string(self.course_id)
+        enrollments = get_user_enrollments(self.course_id).filter(user__is_superuser=False, user__is_staff=False)
+        course_units = list(get_course_units(course_key))
+        data_frame = defaultdict(dict)
+
+        index = 0
+        for enrollment in enrollments:
```

Comment:
```suggestion
        for index, enrollment in enumerate(enrollments):
```

Reply:
I thought so too, but the id should increment at each iteration in the course unit for each course enrollment.

Reply:
Alright so, can we change the index variable name to something more explicit?

Reply:
I renamed this variable to `unique_id`: https://github.com/eduNEXT/platform-plugin-ontask/commit/e17810f5271272bd411ab353d2475b9d2c204660